plugins/cairo-lang-macro/src/types/token.rs

```diff
@@ -0,0 +1,149 @@
+use std::fmt::Display;
+
+use cairo_lang_diagnostics::DiagnosticsBuilder;
+use cairo_lang_filesystem::{
+    ids::{FileKind, FileLongId, VirtualFile},
+    span::TextSpan,
+};
+use cairo_lang_parser::{parser::Parser, utils::SimpleParserDatabase};
+use cairo_lang_syntax::node::TypedSyntaxNode;
+use cairo_lang_syntax::node::{db::SyntaxGroup, SyntaxNode};
+use cairo_lang_utils::Intern;
+use itertools::Itertools;
+use serde::{Deserialize, Serialize};
+
+/// Representation of most atomic struct that is passed within host<->macro communication.
+#[derive(Debug, Default, Clone, Serialize, Deserialize)]
+pub struct Token {
+    span: TextSpan,
+    content: String,
+}
+
+impl Token {
+    pub fn new(content: String, span: TextSpan) -> Self {
+        Self { content, span }
+    }
+
+    pub fn from_syntax_node(db: &dyn SyntaxGroup, node: SyntaxNode) -> Self {
+        Token::new(node.get_text(db), node.span(db))
+    }
+}
+
+/// An abstract stream of Cairo tokens.
+///
+/// This is both input and part of an output of a procedural macro.
+#[derive(Debug, Default, Clone)]
+pub struct TokenStream {
+    pub tokens: Vec<Token>,
```

Comment:
Shouldn't we go for `struct TokenStream -> Vec<enum TokenTree> -> TokenTree::Ident()` already? To be more future safe? 🤔